src/ProjectFileTools.MSBuild/Worksapce.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using Microsoft.Build.Construction;
+using Microsoft.Build.Evaluation;
+using Microsoft.Language.Xml;
+
+namespace ProjectFileTools.MSBuild
+{
+    /// <summary>
+    /// Contains an MSBuild project and logic to extract information from it
+    /// </summary>
+    public class Workspace
+    {
+        private ProjectCollection _collection;
+        private Project _project;
+        private HashSet<string> _containedFiles;
+        private List<FileSystemWatcher> _watchers;
+        private bool _needsReload;
+
+        internal Workspace(string filePath)
+        {
+            _collection = new ProjectCollection();
+            _containedFiles = new HashSet<string>();
```

Comment:
Nit: set a comparer for the HashSet (`StringComparer.Ordinal` seems like a good choice)